packages/apidom-parser/src/parser.ts

```diff
@@ -18,119 +17,124 @@ export type {
 
 export { ParserError };
 
-const ApiDOMParser: stampit.Stamp<ApiDOMParserType> = stampit().init(
-  function ApiDOMParserConstructor() {
-    const adapters: ApiDOMParserAdapter[] = [];
+class ApiDOMParserClass implements ApiDOMParserType {
+  private adapters: ApiDOMParserAdapter[] = [];
 
-    const detectAdapterCandidates = async (source: string) => {
-      const candidates = [];
+  private async detectAdapterCandidates(source: string): Promise<ApiDOMParserAdapter[]> {
+    const candidates = [];
 
-      for (const adapter of adapters) {
-        // eslint-disable-next-line no-await-in-loop
-        if (isFunction(adapter.detect) && (await adapter.detect(source))) {
-          candidates.push(adapter);
-        }
+    for (const adapter of this.adapters) {
+      // eslint-disable-next-line no-await-in-loop
+      if (isFunction(adapter.detect) && (await adapter.detect(source))) {
+        candidates.push(adapter);
       }
+    }
 
-      return candidates;
-    };
+    return candidates;
+  }
 
-    const findAdapter = async (source: string, mediaType: string | undefined) => {
-      if (isString(mediaType)) {
-        return adapters.find((adapter) => {
-          if (!isArray(adapter.mediaTypes)) return false;
+  private async findAdapter(
```

Comment:
Let's allow users to make subclasses and reuse the logic.

```suggestion
  protected async findAdapter(
```